go.mod

```diff
@@ -153,3 +153,8 @@ retract (
 	v0.11.0
 	v0.3.0 // retagged: cd604c1b597558ea045a79c4f80a8c780909801b -> 85653575c7dafb6b06548478ee1dc61ac5905d00
 )
+
+// We are using the forked containerd package version for enabling
+// OCI reference type feature. Once containerd officially releases
+// this feature, we will remove the replacement here.
+replace github.com/containerd/containerd => github.com/nydusaccelerator/containerd v0.0.0-20230331034304-d4d4bdc00ef9
```

Comment:
Maybe we'd better merge this PR into a feature branch with the go.mod `replace` directive. When containerd has released the support, we can safely merge it back to the main branch. Otherwise, we can hardly get rid the experimental feature and use normal containerd

Reply:
Then we can create the feature branch "oci_subject"

Reply:
Fixed, moved the ported codes to `pkg/remote/remotes`.